chunky/src/java/se/llbit/util/interner/Interner.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.llbit.util.interner;
+
+/**
+ * This deduplicates objects similar to `String.intern()`, but for arbitrary objects.
+ */
+public interface Interner<T> {
```

Comment:
I'm not sure about this name. Maybe `ObjectDeduplicator`?

Reply:
`Interner` seems to be the generally used name for this. Google Guava has a similar interface also named [`Interner`](https://guava.dev/releases/22.0/api/docs/com/google/common/collect/Interner.html) and there is also [`String.intern()`](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true#intern--).